### PR TITLE
Default: Allow papyrus growing on default:sand

### PR DESCRIPTION
**Pull request by paramat**
Also:
Mgv5/mgv7 papyrus schematic now has sand nodes as a base
Add beaches of default:sand to mgv5/mgv7 deserts

screenshot_20150830_001944

^ This commit

screenshot_20150830_002323

^ Currently

Sand base nodes blend with beaches and shallow water, it looks better.
In mgv5/mgv7 papyrus grows in shallow water in savanna and desert biomes, allowing papyrus to grow on sand allows growing on the beaches of these biomes without having to place dirt there first. In reality papyrus can grow in wet sandy soil.
The sand around the papyrus base can be considered sand that has got caught-up in, and collected around, the base of the papyrus.

### CODE REVIEW

File: mods/default/functions.lua [hunk 1]
```diff
@@ -152,7 +152,8 @@ end
 function default.grow_papyrus(pos, node)
 	pos.y = pos.y - 1
 	local name = minetest.get_node(pos).name
-	if name ~= "default:dirt_with_grass" and name ~= "default:dirt" then
+	if name ~= "default:dirt_with_grass" and name ~= "default:dirt" and
+			name ~= "group:sand" then
```
Comment (kilbith) [inline mods/default/functions.lua]:
``` Lua
if not name:find("dirt") and not name:find("sand") then
```

Comment (sfan5) [inline mods/default/functions.lua]: * We don't want papyrus to grow on <i>sand</i>worm_spawners or on stuff registered by a tool<i>sand</i>blocks mod. Yes I made those examples up but we didn't ever do this in any other code, why do it here?

* Does this actually work? I don't think the name will be "group:sand" if it's in the sand group. Don't you need to look up the table of registered nodes for that?

**Comment by paramat** [comment]
Updated. Thanks for catching 'name' not working with 'group:sand'.
I changed the added fertile node to be default:sand, assumed to be wet sand near water.
I also added beaches of default:sand to desert biomes so that papyrus can be grown there.

**Comment by sfan5** [comment]
:+1:

**Comment by paramat** [comment]
4d9b595a69988bcd9add832ba4289c5e8285c800

**Pull request closed without merging**